app.py

```diff
@@ -936,18 +982,21 @@ def export_gpx():
             gpx_segment = gpxpy.gpx.GPXTrackSegment()
             gpx_track.segments.append(gpx_segment)
             gps_data = (
-                json.loads(trip["gps"]) if isinstance(trip["gps"], str) else trip["gps"]
+                json.loads(trip["gps"]) if isinstance(
+                    trip["gps"], str) else trip["gps"]
             )
             if gps_data.get("type") == "LineString":
                 for coord in gps_data.get("coordinates", []):
                     if isinstance(coord, list) and len(coord) >= 2:
                         lon, lat = coord[0], coord[1]
-                        gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                        gpx_segment.points.append(
+                            gpxpy.gpx.GPXTrackPoint(lat, lon))
             elif gps_data.get("type") == "Point":
                 coord = gps_data.get("coordinates", [])
                 if isinstance(coord, list) and len(coord) >= 2:
                     lon, lat = coord[0], coord[1]
-                    gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                    gpx_segment.points.append(
+                        gpxpy.gpx.GPXTrackPoint(lat, lon))
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating duplicate GPX point creation</b></div>
<div id="fix">

Consider consolidating the duplicate GPX track point creation logic into a helper function to reduce code duplication.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -984,18 +984,22 @@
  def add_gpx_point(segment, lat, lon):
      segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))

  if gps_data.get("type") == "LineString":
      for coord in gps_data.get("coordinates", []):
          if isinstance(coord, list) and len(coord) >= 2:
              lon, lat = coord[0], coord[1]
 -            gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +            add_gpx_point(gpx_segment, lat, lon)
  elif gps_data.get("type") == "Point":
      coord = gps_data.get("coordinates", [])
      if isinstance(coord, list) and len(coord) >= 2:
          lon, lat = coord[0], coord[1]
 -        gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/realronaldrump/everystreet-new/pull/99#issuecomment-2549779985>#22e49e</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged